### ISSUE TITLE
Amazon EventBridge

### ISSUE THREAD

**Issue by ewbankkit** [hashicorp/terraform-provider-aws#9330]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Amazon EventBridge [announced](https://aws.amazon.com/about-aws/whats-new/2019/07/introducing-amazon-eventbridge/).

At the AWS API level these are additions to the existing CloudWatch Events API.

### New or Affected Resource(s)

Sticking with the existing naming convention of `aws_cloudwatch_event_`

#### New Resource

* aws_cloudwatch_event_event_bus

#### Modified Resources

* [`aws_cloudwatch_event_permission`](https://www.terraform.io/docs/providers/aws/r/cloudwatch_event_permission.html)
* [`aws_cloudwatch_event_rule`](https://www.terraform.io/docs/providers/aws/r/cloudwatch_event_rule.html)
* [`aws_cloudwatch_event_target`](https://www.terraform.io/docs/providers/aws/r/cloudwatch_event_target.html)

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

Requires:
* https://github.com/terraform-providers/terraform-provider-aws/pull/9305

**Comment by ebekker** [comment]
I would just stick the simpler `aws_cloudwatch_event_bus` -- the resource is a _bus_ in the CW Event system.

**Comment by ewbankkit** [comment]
@ebekker Yes, I tend to agree. Of these CloudWatch resources, only [`aws_cloudwatch_metric_alarm`](https://www.terraform.io/docs/providers/aws/r/cloudwatch_metric_alarm.html) matches the [naming guidelines](https://github.com/terraform-providers/terraform-provider-aws/blob/master/.github/CONTRIBUTING.md#new-resource).

**Comment by tanmng** [comment]
@ewbankkit given the change in commit `8295926` in #10256 by @alexanderkalach, I think this issue can no longer be closed by that PR, and we will need separate effort to handle AWS partner event bus. Is that the case? 
If so, can we proceed with the result of the vote and start getting an implementation on this?

**Comment by ewbankkit** [comment]
@tanmng Yes, #10256 is just the initial work needed. There will be additional PRs to complete functionality.

**Comment by peramnaveen29** [comment]
@ewbankkit I have a noticed a difference in the way rules are being applied on the cloudwatch and Eventbridge. 

We came across a use case where we had to use wildcard in event patterns arguments. Below example.

`{
  "detail": {
    "state": [
      "SUCCEEDED",
      "Succeeded"
    ],
    "jobName": [
      **"*test*"**
    ]
  },
  "detail-type": [
    "Glue Job State Change"
  ],
  "source": [
    "aws.glue"
  ]
}`

The above event pattern with wildcard (highlited) is not supported for AWS cloudwatch events rules. But interestingly wildcard is working when you add the same in AWS EventBridge Events rules. Not sure how this is implemented by AWS but this works.

Maybe this is something to be looked upon while implementing EventBridge resource with terraform.

**Comment by breathingdust** [comment]
Hi Everyone! :wave:

We recognize the significant community interest in this service area and want to provide an update about its status. 

Because there is significant overlap between EventBridge and Cloudwatch we’ll need to take a step back and spend a bit of time considering design. 

Once we’ve finished our research, we’ll post an update to the issue detailing the next steps.

We appreciate all the contributions and feedback thus far!

**Comment by claranet-barney** [comment]
Hey @breathingdust / @ewbankkit Any news on how this is coming along?  Not hassling just wondering if there's been any movement on it or any plans in progress?

**Comment by breathingdust** [comment]
Hey @claranet-barney! 👋 

Nothing to report as yet, we'll need to work through this quarters [ROADMAP](https://github.com/terraform-providers/terraform-provider-aws/blob/master/ROADMAP.md) items first. It's still definitely on our list and I would expect to see it released sometime this summer. Thanks for your patience!

**Comment by kurtmaile** [comment]
Hi,

Is event bridge support still outstanding?

Cheers

**Comment by mwarkentin** [comment]
In the meanwhile, you can do the cloudformation hack:

```hcl
resource "aws_cloudformation_stack" "eventbridge_bus" {
  name = "eventbridge-bus"

  template_body = <<EOF
Resources:
  EventBus:
    Type: AWS::Events::EventBus
    Properties:
      Name: bus-name
EOF
}
```

**Comment by DylanBohlender** [comment]
It looks like this has been added to the [August - October ROADMAP](https://github.com/terraform-providers/terraform-provider-aws/blob/3713ea9f8ce8f3bb1fd572beed8b1fa1967af7ff/ROADMAP.md) file as the first item, which is awesome!

Is there a rough timeline for when this will be implemented? I suspect that the design research phase likely extended the timelines here, but any visibility into progress would be greatly appreciated. The CF template hack above is a useful workaround, but I suspect I'm not alone in feeling icky every time I'm writing inline CF templates in lieu of HCL.

**Comment by maryelizbeth** [comment]
Hey @DylanBohlender! Right now, we don't have a release date for EventBridge support. The team is actively researching and will begin implementation once that is complete. Once we've decided on the design of this feature, we'll begin work and you'll see it added to a release milestone. We'll come back to this issue with updates as they're available.

**Comment by smailc** [comment]
Any news on EventBridge support?

**Comment by breathingdust** [comment]
Hi all! 👋 

Just an update that we have finished the service design and plan to start work this week, so we are on track to have this out by the end of October. Again, we appreciate your patience!

**Comment by breathingdust** [comment]
Initial scope of EventBridge support has been completed as of [v3.14.1](https://github.com/hashicorp/terraform-provider-aws/releases/tag/v3.14.1) 🎉
